### PR TITLE
OSDOCS-998 declarative configs for networking

### PR DESCRIPTION
**Pull request by mikemckiernan**
/hold

This is a "feeler PR" to provoke ideas.  I started with Multus because that is an area that is currently documented as imperative with `oc edit`.

This is in the wrong loc for a broad title like "Networking and GitOps"--either the title needs to change or the loc and breadth need to increase.

https://deploy-preview-31207--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/networking-gitops.html

### CODE REVIEW

File: _topic_map.yml [hunk 1]
```diff
@@ -854,6 +854,8 @@ Topics:
     File: remove-additional-network
   - Name: Assigning a secondary network to a VRF
     File: assigning-a-secondary-network-to-a-vrf
+  - Name: Networking and GitOps
```
Comment (jboxman) [inline _topic_map.yml:857]: But then it follows that every section of the OCP docs ought to have a _\<section> and GitOps_ section for consistency.

Maybe every section should do that; I thought GitOps might be more of a transparent thing, though.

File: modules/nw-multus-network-json.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * networking/multiple_newtorks/networking-gitops.adoc
+
+[id="nw-multus-network-json_{context}"]
+= Configuration for network attachment definition
+
+The following sections provide reference information about the network plugins that you can use when you create a JSON-formatted network attachment definition.
```
Comment (jboxman) [inline modules/nw-multus-network-json.adoc:8]: s/plugins/plug-ins/

File: modules/nw-multus-gitops.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// * networking/multiple_networks/networking-gitops.adoc
+
+[id="nw-multus-gitops_{context}"]
+= Multiple networks and GitOps
+
+Typically, managing an additional network is performed imperatively by editing the `cluster` object for the Cluster Network Operator with the `oc edit networks.operator.openshift.io cluster` command and adding a `.spec.additionalNetworks` field.
```
Comment (jboxman) [inline modules/nw-multus-gitops.adoc:8]: I've gone both ways, but ultimately I now drop the leading `.` for YAML fields.

File: modules/nw-multus-gitops.adoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+To support GitOps, the best practice is to begin by creating a network attachment definition object instead of modifying the `cluster` object.
+
+== Limitations
+
+* After you add a network attachment definition and a pod uses the network attachment, modifying the network attachment definition does not trigger the pod to restart. As a result, the pod continues to run with the previous definition.
```
Comment (jboxman) [inline modules/nw-multus-gitops.adoc:14]: I think this is true in general, though. A pod must be deleted and re-created to benefit from updates to an existing NAD.

File: modules/nw-multus-network-json.adoc [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+== Configuration for bridge
+
+include::nw-multus-bridge-object.adoc[tag=json-ref]
```
Comment (jboxman) [inline modules/nw-multus-network-json.adoc:12]: Would need to confirm that we're actually allowed to do this, both the include and the tag style.

**Review by jboxman — COMMENTED** [review]
I left a few harried comments.

File: _topic_map.yml (hunk 1 above)
Comment (mikemckiernan) [reply _topic_map.yml:857]: Let's discuss--and maybe get a little direction!  My thinking was that we'd need to either change what we had (that didn't work, it was terrible) or add new info (this PR).
Thought no. 3 is to just remove the `oc edit` step from the existing content.

File: modules/nw-multus-network-json.adoc (hunk 5 above)
Comment (mikemckiernan) [reply modules/nw-multus-network-json.adoc:12]: I suppose that I could "single source" it--duplicate the source every single time.  Seriously, I have no strong feeling and this PR is really to trigger conversation.

Comment (jboxman) [reply modules/nw-multus-network-json.adoc:12]: @mikemckiernan, oh, I thought in that case you could just put the includes in the assembly instead?

**Pull request closed without merging**